File: 6labs/ts/TextLine.ts
import { ICaseChanger } from "./ICaseChanger";

export class TextLine implements ICaseChanger {
    private content: string;

    constructor(text: string) {
        this.content = text;
    }

    toUpperCase(): string {
        return this.content.toUpperCase();
    }

    toLowerCase(): string {
        return this.content.toLowerCase();
    }

    capitalizeWords(): string {
        return this.content
            .split(' ')
            .map(word => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
            .join(' ');
    }

    getContent(): string {
        return this.content;
    }

    setContent(newContent: string): void {
        this.content = newContent;
    }
}

File: 6labs/c#/TextLine.cs
public class TextLine : ICaseChanger
{
    private string content;

    public TextLine(string text)
    {
        content = text;
    }

    public string ToUpperCase() => content.ToUpper();
    public string ToLowerCase() => content.ToLower();
    public string CapitalizeWords()
    {
        return string.Join(" ", content
            .Split(' ')
            .Select(word => char.ToUpper(word[0]) + word.Substring(1).ToLower()));
    }

    public string GetContent() => content;
    public void SetContent(string newContent) => content = newContent;
}
